vb-fe/src/services/Socket.ts

```diff
@@ -0,0 +1,44 @@
+import socketIOClient from "socket.io-client";
+
+import { TokenStorageKeys } from "util/Http";
+import { Room } from "util/Room";
+import { SOCKET_ENDPOINT, SocketError, SocketMessage } from "util/Socket";
+
+export const socket = socketIOClient(SOCKET_ENDPOINT, {
+  transportOptions: {
+    polling: {
+      extraHeaders: {
+        [TokenStorageKeys.AT]: localStorage.getItem(TokenStorageKeys.AT),
+      },
+    },
+  },
+});
+
+/**
+ * Listener events and callbacks
+ */
+export const subscribeTo = {
+  joinSuccess: (cb: (data: Room) => unknown): void => {
```

Comment:
so here, i was thinking of just putting a single event for success. `socket-success`

The callback of this success will have the interface
```
{
   type: "JOIN_ROOM" | "PLAY_TRACK" .... etc,
   data: {..whatever}
}
```

And then have a `switch(type) {}` similar to redux.

Same for the error handler.

What do you think?

Reply:
Hmm, not a bad idea if we're going to have several socket success events coming to the FE where we need data.  Any idea what other data are we going to need returned to the `FE` for these events?

Reply:
Otherwise it would probably be best to keep this singular just so we always know the type.  but if there are several events we need some sort of success data then I'm game

Reply:
Umm, I'm not sure. We can encapsulate everything in `data` and send it, like its happening right now

Reply:
Well rn I can't think of any but I was just wondering maybe in the future for scalability

Reply:
Yeah that sounds fine, let's implement that when we get a second event that will use it though, until then this should be fine

Reply:
Alright sounds good